pkg/cloudproduct/generic/generic_test.go

```diff
@@ -0,0 +1,221 @@
+			}),
+		},
+		"SafeToEvict: OnUpgrade, no incoming labels/annotations": {
+			featureFlags: "SafeToEvict=true",
+			safeToEvict:  agonesv1.EvictionSafeOnUpgrade,
+			pod:          emptyPodAnd(func(*corev1.Pod) {}),
+			wantPod: emptyPodAnd(func(pod *corev1.Pod) {
+				pod.ObjectMeta.Annotations[agonesv1.PodSafeToEvictAnnotation] = agonesv1.False
+				pod.ObjectMeta.Labels[agonesv1.SafeToEvictLabel] = agonesv1.True
+			}),
+		},
+		"SafeToEvict: Never, no incoming labels/annotations": {
+			featureFlags: "SafeToEvict=true",
+			safeToEvict:  agonesv1.EvictionSafeNever,
+			pod:          emptyPodAnd(func(*corev1.Pod) {}),
+			wantPod: emptyPodAnd(func(pod *corev1.Pod) {
+				pod.ObjectMeta.Annotations[agonesv1.PodSafeToEvictAnnotation] = agonesv1.False
+				pod.ObjectMeta.Labels[agonesv1.SafeToEvictLabel] = agonesv1.False
+			}),
+		},
+		"SafeToEvict: Always, incoming labels/annotations": {
+			featureFlags: "SafeToEvict=true",
+			safeToEvict:  agonesv1.EvictionSafeAlways,
+			pod: emptyPodAnd(func(pod *corev1.Pod) {
+				pod.ObjectMeta.Annotations[agonesv1.PodSafeToEvictAnnotation] = "just don't touch, ok?"
+				pod.ObjectMeta.Labels[agonesv1.SafeToEvictLabel] = "seriously, leave it"
+			}),
+			wantPod: emptyPodAnd(func(pod *corev1.Pod) {
+				pod.ObjectMeta.Annotations[agonesv1.PodSafeToEvictAnnotation] = "just don't touch, ok?"
+				pod.ObjectMeta.Labels[agonesv1.SafeToEvictLabel] = "seriously, leave it"
+			}),
+		},
+		"SafeToEvict: OnUpgrade, incoming labels/annotations": {
+			featureFlags: "SafeToEvict=true",
+			safeToEvict:  agonesv1.EvictionSafeOnUpgrade,
+			pod: emptyPodAnd(func(pod *corev1.Pod) {
+				pod.ObjectMeta.Annotations[agonesv1.PodSafeToEvictAnnotation] = "better not touch"
+				pod.ObjectMeta.Labels[agonesv1.SafeToEvictLabel] = "not another one"
+			}),
+			wantPod: emptyPodAnd(func(pod *corev1.Pod) {
+				pod.ObjectMeta.Annotations[agonesv1.PodSafeToEvictAnnotation] = "better not touch"
+				pod.ObjectMeta.Labels[agonesv1.SafeToEvictLabel] = "not another one"
+			}),
+		},
+		"SafeToEvict: Never, incoming labels/annotations": {
+			featureFlags: "SafeToEvict=true",
+			safeToEvict:  agonesv1.EvictionSafeNever,
+			pod: emptyPodAnd(func(pod *corev1.Pod) {
+				pod.ObjectMeta.Annotations[agonesv1.PodSafeToEvictAnnotation] = "a passthrough"
+				pod.ObjectMeta.Labels[agonesv1.SafeToEvictLabel] = "or is it passthru?"
+			}),
+			wantPod: emptyPodAnd(func(pod *corev1.Pod) {
+				pod.ObjectMeta.Annotations[agonesv1.PodSafeToEvictAnnotation] = "a passthrough"
+				pod.ObjectMeta.Labels[agonesv1.SafeToEvictLabel] = "or is it passthru?"
+			}),
+		},
+	} {
+		t.Run(desc, func(t *testing.T) {
+			err := runtime.ParseFeatures(tc.featureFlags)
+			assert.NoError(t, err)
+
+			err = (&generic{}).SetEviction(tc.safeToEvict, tc.pod)
+			assert.NoError(t, err)
+			assert.Equal(t, tc.wantPod, tc.pod)
+		})
+	}
+}
+
+func TestGameServerPodAutoscalerAnnotations(t *testing.T) {
```

Comment:
Similarly, were there any changes from the deleted `TestGameServerPodAutoscalerAnnotations` function in `pkg/apis/agones/v1/gameserver_test.go`?

Reply:
It was modified to shift packages but is otherwise the same.